WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentsViewModel.kt

```diff
@@ -154,30 +176,37 @@ class NewSiteCreationSegmentsViewModel
             items.add(segment)
         }
     }
+}
 
-    data class UiState(
-        val showError: Boolean,
-        val showContent: Boolean,
-        val items: List<ItemUiState>
-    )
+interface SegmentsUiState
+data class SegmentsContentUiState(val items: List<SegmentsItemUiState>) : SegmentsUiState
+data class SegmentsErrorUiState constructor(val titleResId: Int, val subtitleResId: Int? = null) : SegmentsUiState {
+    companion object {
+        fun createGenericErrorUiState() = SegmentsErrorUiState(
```

Comment:
Can we change this to a `val`?

```suggestion
        val genericErrorUiState = SegmentsErrorUiState(
```

Reply:
👍 -> I've used the proposed solution from the other comment.